Backend/AMQPBackend.php

```diff
@@ -212,29 +216,28 @@ public function handle(MessageInterface $message, EventDispatcherInterface $disp
     {
         $event = new ConsumerEvent($message);
 
+        /** @var AmqpMessage $amqpMessage */
+        $amqpMessage = $message->getValue('AMQMessage');
+
         try {
             $dispatcher->dispatch($message->getType(), $event);
 
-            $message->getValue('AMQMessage')->delivery_info['channel']->basic_ack($message->getValue('AMQMessage')->delivery_info['delivery_tag']);
+            $this->consumer->acknowledge($amqpMessage);
 
             $message->setCompletedAt(new \DateTime());
             $message->setState(MessageInterface::STATE_DONE);
         } catch (HandlingException $e) {
             $message->setCompletedAt(new \DateTime());
             $message->setState(MessageInterface::STATE_ERROR);
 
-            $message->getValue('AMQMessage')->delivery_info['channel']->basic_ack($message->getValue('AMQMessage')->delivery_info['delivery_tag']);
+            $this->consumer->acknowledge($amqpMessage);
 
             throw new HandlingException('Error while handling a message', 0, $e);
         } catch (\Exception $e) {
             $message->setCompletedAt(new \DateTime());
             $message->setState(MessageInterface::STATE_ERROR);
 
-            if (true === $this->recover) {
-                $message->getValue('AMQMessage')->delivery_info['channel']->basic_recover($message->getValue('AMQMessage')->delivery_info['delivery_tag']);
-            } elseif (null !== $this->deadLetterExchange) {
-                $message->getValue('AMQMessage')->delivery_info['channel']->basic_reject($message->getValue('AMQMessage')->delivery_info['delivery_tag'], false);
-            }
+            $this->consumer->reject($amqpMessage, $this->recover);
```

Comment:
It would be good if you could thoroughly review this part. I might've broken it cuz I did not quite get the purpose of basic_recover.

Reply:
I didn't write the package that well and can't test it (I don't think I even use it). Maybe someone in @sonata-project/contributors  can?